pkg/search/postgres/store.go

```diff
@@ -178,6 +178,32 @@ func (s *GenericStore[T, PT]) Walk(ctx context.Context, fn func(obj PT) error) e
 	return nil
 }
 
+// WalkByQuery iterates over all the objects scoped by the query and applies the closure.
+func (s *GenericStore[T, PT]) WalkByQuery(ctx context.Context, query *v1.Query, fn func(obj PT) error) error {
```

Comment:
Can `Walk` be implemented by this?
```go
func (s *GenericStore[T, PT]) Walk(ctx context.Context, fn func(obj PT) error) error {
	return s.WalkWithQuery*ctx, search.EmptyQuery(), fn) 
}
```

Reply:
Done, but added an intermediate func so we can distinguish between Walk and WalkByQuery metrics